extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/health/KafkaHealthCheck.java

```diff
@@ -0,0 +1,56 @@
+package io.quarkus.kafka.client.health;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.enterprise.context.ApplicationScoped;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.AdminClientConfig;
+import org.apache.kafka.common.Node;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+import org.eclipse.microprofile.health.Readiness;
+
+@Readiness
+@ApplicationScoped
+public class KafkaHealthCheck implements HealthCheck {
+    @ConfigProperty(name = "kafka.bootstrap.servers", defaultValue = "localhost:9092")
+    private String bootstrapServer;
+
+    private AdminClient client;
+
+    @PostConstruct
+    void init() {
+        Map<String, Object> conf = new HashMap<>();
+        conf.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServer);
+        conf.put(AdminClientConfig.REQUEST_TIMEOUT_MS_CONFIG, "5000");
+        client = AdminClient.create(conf);
```

Comment:
How about creating the client (and closing it) only in `call()`? That way we don't hold this client for the entire lifetime of the application, also if it's actually not needed any longer once the container has received traffic and the readiness probe thus shouldn't matter any longer.

Reply:
Typically, a readiness probe will be called every n seconds (5s for example) so I'm concern of the cost of recreating a client each time.

Reply:
I looked into the Vert.x Kafka code and they hold the KafkaClient (inside the Sender) for the entire life of the application so I don't think something is wrong here. But I'm OK to reniew it each time if you prefer. Just tell me :)

Reply:
Yeah, seems fine as is then. Thanks for checking with Vert.x.